src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -763,17 +763,18 @@ protected virtual GasConsumed Refund(Transaction tx, BlockHeader header, IReleas
                 spentGas -= unspentGas;
                 operationGas -= unspentGas;
                 spentGas = Math.Max(spentGas, floorGas);
+                var unspentGasRefund = tx.GasLimit - spentGas;
```

Comment:
Can you write the reasoning for the PR in description? It is probably fine but hard to follow the intention.
We now have `unspentGas` and `unspendGasRefund` which is confusing. Maybe you can add comments about it?

Reply:
I agree with this comment